apps/public-reference/pages/applications/preferences/select.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+4.1 Preferences Introduction
+Instructions on how preferences work and their value
+*/
+import Link from "next/link"
+import Router from "next/router"
+import { Button, ErrorMessage, FormCard, ProgressNav, t } from "@bloom-housing/ui-components"
+import FormsLayout from "../../../layouts/forms"
+import { useForm } from "react-hook-form"
+import { AppSubmissionContext } from "../../../lib/AppSubmissionContext"
+import ApplicationConductor from "../../../lib/ApplicationConductor"
+import FormStep from "../../../src/forms/applications/FormStep"
+import { useContext, useMemo, useState } from "react"
+
+export default () => {
+  const { conductor, application, listing } = useContext(AppSubmissionContext)
+  const currentPageStep = 4
+
+  const preferenceOptions = ["liveIn", "workIn"]
+
+  /* Form Handler */
+  const { getValues, register, handleSubmit, errors, setValue, trigger } = useForm()
+  const onSubmit = (data) => {
+    new FormStep(conductor).save({ preferences: data })
+    if (data.none) {
+      conductor.routeToNextOrReturnUrl("/applications/preferences/general")
+    } else {
+      conductor.completeStep(4)
+      conductor.sync()
+      conductor.routeToNextOrReturnUrl("/applications/review/demographics")
+    }
+  }
+
+  return (
+    <FormsLayout>
+      <FormCard header="LISTING">
+        <ProgressNav
+          currentPageStep={currentPageStep}
+          completedSteps={application.completedStep}
+          labels={["You", "Household", "Income", "Preferences", "Review"]}
+        />
+      </FormCard>
+
+      <FormCard>
+        <p className="form-card__back">
+          <strong>
+            <Link href="/applications/financial/income">
+              <a>{t("t.back")}</a>
+            </Link>
+          </strong>
+        </p>
+
+        <div className="form-card__lead border-b">
+          <h2 className="form-card__title is-borderless">{t("application.preferences.title")}</h2>
+
+          <p className="field-note mt-5">{t("application.preferences.preamble")}</p>
+        </div>
+
+        <form className="" onSubmit={handleSubmit(onSubmit)}>
+          <div className="form-card__group px-0 pb-3">
+            <p className="field-note">{t("application.preferences.selectBelow")}</p>
+          </div>
+
+          {preferenceOptions.map((option) => {
+            const [showMore, setShowMore] = useState(false)
```

Comment:
Using a hook inside the render is generally considered to be a mistake in React. One of the rules of hooks is that the number of hooks that you render needs to be invariant every time the component renders (e.g. can't change based on conditional logic), so you want to avoid burying hooks deep in the render logic where someone else might come along later and add conditional rendering - it's much more clear to have them at the top of your component function. This is doubly true for using them within an iteration - in this case `preferenceOptions` is a fixed length array, which is why it works, but if it was ever made variable (e.g. we loaded options asynchronously) this logic could cause problems. 

To fix this, you should either move the state up to the top of the function:

```typescript
const [showMore, setShowMore] = useState({})
...
// Read from the state for a specific option
showMore[option]

const toggleShowMoreForOption = (option) => setShowMore({ ...showMore, [option]: !showMore[option] })
```

Or refactor the iteration render into a subcomponent that can keep track of its own `showMore` state internally and take `option` as a prop.

Reply:
@bencpeters I'll move it out to a subcomponent, as it does need to maintain separate state for each item.